### DISCUSSION TITLE
JSON serialize Date

### DISCUSSION THREAD

**Discussion by JolloDede** [joshnuss/svelte-local-storage-store#262] [Q&A] [answered]
I have a project where i am using your library. I ran into problems when serializing and deserialising dates.
I know the JSON format doesnt have a dateformat [Stackoverflow discussion](https://stackoverflow.com/questions/10286204/what-is-the-right-json-date-format). Javascript on the otherhand has a format `2012-04-23T18:25:43.511Z`

Is it possible to have something that parses the date, in the js format, when calling reading from the localstorage.
Like:
```Javascript
dateTimeReviver = function (key, value) {
    var a;
    if (typeof value === 'string') {
        a = /\/Date\((\d*)\)\//.exec(value);
        if (a) {
            return new Date(+a[1]);
        }
    }
    return value;
}

JSON.parse(somejsonstring, dateTimeReviver);
```

[Stackoverflow discussion](https://stackoverflow.com/questions/4511705/how-to-parse-json-to-receive-a-date-object-in-javascript)

Or is there someway i am missing to get the date parsed

**Comment by joshnuss** [comment, marked as answer]
Hi @JolloDede

JSON does not support dates.
But you can use `devalue` instead. Just specify an option for `serializer`, for example: `serializer: devalue`.
